lib/elixir/src/elixir_env.erl

```diff
@@ -85,10 +85,20 @@ reset_unused_vars(#elixir_ex{unused={_Unused, Version}} = S) ->
   S#elixir_ex{unused={#{}, Version}}.
 
 check_unused_vars(#elixir_ex{unused={Unused, _Version}}, E) ->
-  [elixir_errors:file_warn(Meta, E, ?MODULE, {unused_var, Name, Overridden}) ||
+  [elixir_errors:file_warn(calculate_span(Name, Meta), E, ?MODULE, {unused_var, Name, Overridden}) ||
     {{{Name, nil}, _}, {Meta, Overridden}} <- maps:to_list(Unused), is_unused_var(Name)],
   E.
 
+calculate_span(Name, Meta) ->
```

Comment:
Nitpick, we have the Meta as first argument almost everywhere. Can you keep it consistent please? :)